timemachine/cpp/src/bound_potential.cu

```diff
@@ -58,15 +55,27 @@ void BoundPotential::execute_host(
     }
 };
 
+void BoundPotential::set_params(const std::vector<double> &params) {
+    if (params.size() > 0) {
```

Comment:
For the host code, don't think we want to preserve this logic of handling 0 like this. Instead, it should just do the check of `params.size() != buffer_size`.

Reply:
Unless there is a reason to pass empty parameters, it seems like this branch could be removed in both the host and device code. For device code, it avoids a trivial kernel launch, but I'm not sure we would ever trigger this intentionally anyway?